lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,10 +37,20 @@ def to_native(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            try:
+                get_discussion_course_or_404(CourseKey.from_string(course_id), request.user)
```

Comment:
@nasthagiri 
The `CourseEnrollment` model creates `course_overview` property from its course `CourseOverview.get_from_id(self.course_id)` which in turns loads course_overview from cache and if not present then calls modulestore and creates `course_overview = cls._create_from_course(course)`
By looking at it; course_overview is being created from course and cashed. I was wondering, why did not we cache complete course and serialized course_overview from that course cashed object to improve performance?